croncat/src/config/mod.rs

```diff
@@ -7,13 +7,32 @@ use config::Config;
 use serde::{Deserialize, Serialize};
 use std::path::Path;
 
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ChainConfigSource {
+    pub rpc_endpoint: String,
+    pub wsrpc_endpoint: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ChainConfigFile {
+    pub denom: String,
+    pub prefix: String,
+    pub chain_id: String,
+    pub sources: Vec<ChainConfigSource>,
+    pub grpc_endpoint: String,
```

Comment:
@SeedyROM why is this outside `ChainConfigSource`?
The 3 RPC, GRPC & Websockets all map to 1 provider, so this should be part of the group ChainConfigSource

Reply:
Right now the grpc signer is global regardless of how many block sources we have. Should each source be it's own agent task loop? I'm also not a fan of this code either, but it's the only way to not break all the client interaction code (since the they all take in the config object in it's entirety).